lua/line-number-change-mode/init.lua

```diff
@@ -2,28 +2,39 @@ local M = {}
 
 function M.setup(opts)
    opts = opts or {}
-
-   local group = vim.api.nvim_create_augroup("LineNumberChangeMode", {
-      clear = true
+   local va = vim.api
+   local group = va.nvim_create_augroup("LineNumberChangeMode", {
+      clear = true,
    })
 
-   vim.api.nvim_create_autocmd('ModeChanged', {
+   local function set_hl_for_mode(mode)
+      if opts.mode[mode] then
+         va.nvim_set_hl(0, "CursorLineNr", opts.mode[mode])
+
+         if mode == "c" then
+            vim.cmd.redraw()
+         end
+      end
+   end
+
+   va.nvim_create_autocmd("ModeChanged", {
       group = group,
       callback = function()
          local new_mode = vim.v.event.new_mode
-         local line_number_map = {
-            i = opts.i and opts.i or nil,
-            n = opts.n and opts.n or nil,
-            no = opts.no and opts.no or nil,
-            R = opts.R and opts.R or nil,
-            v = opts.n and opts.v or nil,
-            V = opts.V and opts.V or nil,
-         }
-
-         if line_number_map[new_mode] ~= nil and opts[new_mode] ~= nil then
-            vim.api.nvim_set_hl(0, 'CursorLineNr', line_number_map[new_mode])
+
+         if opts.debug then
+            print(new_mode)
          end
-      end
+
+         set_hl_for_mode(new_mode)
+      end,
+   })
+
+   va.nvim_create_autocmd("VimEnter", {
+      group = group,
+      callback = function()
+         set_hl_for_mode(va.nvim_get_mode().mode)
+      end,
```

Comment:
The problem with the `VimEnter` event is the plugin may not get that event it's lazy loaded in some ways (e.g.` event = 'VeryLazy',` which is common for "nice to have" plugins).

I think it's cleaner to just call `set_hl_for_mode(va.nvim_get_mode().mode)` as part of `setup` rather than tying it to another event. That way it will always be called and the user has control over when that happens. If they lazy load via `VeryLazy`, that's fine, it'll still be called. If they want it to load on `VimEnter` they can set `event = 'VimEnter'` in the plugin spec and have it load a little earlier (this is all assuming Lazy.nvim is being used)

Reply:
That makes sense.  I can change it back.